### PR TITLE
Align styles with vaadin-tab

### PR DESCRIPTION
**Pull request by manolo**
Connects to https://github.com/vaadin/vaadin-core/issues/105

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-item/5)

### CODE REVIEW

**Comment by platosha** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[vaadin-item.html, line 15 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIj5zJQKOSBI-O638t:-KxIj5zJQKOSBI-O638u:b-xn41vi) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L15)):*
> ```HTML
>     <style>
>       :host {
>         display: flex;
> ```

Well, I read this as: `:host` is a flex container for children.

Then I look what the flex children are in the template: they are `<style>` and `<slot>`.

Having a flex container over `<style>` and `<slot>` does not make sense, right?

---

*[vaadin-item.html, line 16 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIjjwpHa8rC2ft3cFd:-KxIjjwpHa8rC2ft3cFe:b-dex8rd) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L16)):*
> ```HTML
>       :host {
>         display: flex;
>         align-items: center;
> ```

Same concern as above. Alignment settings for a flex container which does not make sense.

---

*[vaadin-item.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIiaOQoj2vHt-7orMg:-KxIiaOQoj2vHt-7orMh:bwb1qte) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L20)):*
> ```HTML
>         opacity: 0.7;
>         padding: 0.3em 1em;
>         box-sizing: border-box;
> ```

Why `border-box` here?

---

*[vaadin-item.html, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIibmjyB5Z1PHNkans:-KxIibmjyB5Z1PHNkant:b5ida7a) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L21)):*
> ```HTML
>         padding: 0.3em 1em;
>         box-sizing: border-box;
>         flex-shrink: 0;
> ```

Duplicate of builtin style. FYI: the builtin style should probably be moved out too, see a comment below.

---

*[vaadin-item.html, line 43 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIkcNjAsWzMFk0Ng57:-KxIkcNjAsWzMFk0Ng58:b2tco10) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L43)):*
> ```HTML
>     <style>
>       :host {
>         display: inline-block;
> ```

OK probably does not affect much, but still have a question.

Default theme makes `display: flex`, which acts as a block for the parent.

Without default theme, it would be `inline-block`, which acts as an inline in the parent.

Why there is such a difference?

---

*[vaadin-item.html, line 45 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIhdG2LhEPH_3-O9NW:-KxIhdG2LhEPH_3-O9NX:by6nu69) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L45)):*
> ```HTML
>         display: inline-block;
>         cursor: pointer;
>         flex-shrink: 0;
> ```

OK maybe not a blocker, but not sure if this is appropriate here.

`flex-shrink` changes are only needed if a parent is a flex container.

If this is because of `<vaadin-tabs>` having a flex layout for slotted children, then should we rather move this fix to `<vaadin-tabs>` slotted styles?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[vaadin-item.html, line 15 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIj5zJQKOSBI-O638t:-KxIuCLYsp3L8vt30GPh:bw0imaa) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L15)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Well, I read this as: `:host` is a flex container for children.

Then I look what the flex children are in the template: they are `<style>` and `<slot>`.

Having a flex container over `<style>` and `<slot>` does not make sense, right?
</blockquote></details>

@platosha  all those styles have been moved from vaadin-tab without modification, most of them were set there on this commit https://github.com/vaadin/vaadin-tabs/commit/9a2207fe  @jouni can you answer Anton ? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5)*

**Comment by jouni** [comment]
I think we discussed with @tomivirkki that `vaadin-item` should not have any of the unnecessary features like selected, focused or disabled, and those would be handled by `ListMixin`. In that case, all of the styles should also come from parent list element.

An extending element, like `vaadin-tab` can choose to style the extra states like `[selected]` and `[disabled]`, if the parent scoped styles are not enough.

**Comment by manolo** [comment]
So we don't offer CSS api for those statuses in vaadin-items ?

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-:-KxIyLGC0szEmyOlNBSg:bihuc4)*

File: vaadin-item.html [hunk 1]
```diff
@@ -12,10 +12,25 @@
...
-        padding: 4px;
-        white-space: nowrap;
-        text-overflow: ellipsis;
+        display: flex;
+        align-items: center;
+        outline: none;
+        opacity: 0.7;
+        padding: 0.3em 1em;
         box-sizing: border-box;
+        flex-shrink: 0;
```
Comment (jouni) [inline vaadin-item.html]: - Remove display and align-items, let’s set those in the core/default styles (not the theme).
- Remove outline and opacity, which are only necessary when used inside a list element.
- Remove padding and box-sizing, let’s keep the default styles to bare minimum for the independent item.
- Remove flex-shrink, it’s only needed for `vaadin-tab`.

File: vaadin-item.html [hunk 2]
```diff
@@ -12,10 +12,25 @@
...
+        opacity: 1;
+      }
+
+      :host([focus-ring]) {
+        background-color: rgba(0, 0, 0, 0.1);
+      }
+
+      :host([disabled]) {
+        opacity: 0.3;
       }
```
Comment (jouni) [inline vaadin-item.html]: Remove all state based styles.

File: vaadin-item.html [hunk 3]
```diff
@@ -25,8 +40,9 @@
   <template>
     <style>
       :host {
-        display: block;
+        display: inline-block;
         cursor: pointer;
+        flex-shrink: 0;
```
Comment (jouni) [inline vaadin-item.html]: - Use `display: flex;` and `align-items: center;`
  - I’m not 100% sure if these are already good to have, but I’m expecting us to use flexbox once we add the planned slots (icon/prefix, label/description and badge/action)
- Remove flex-shrink (only needed for `vaadin-tab`).
- Remove cursor, only needed when used inside a list element.

Comment (jouni) [reply vaadin-item.html]: Below: remove the disabled state related styles.

**Comment by manolo** [comment]
Review status: 0 of 1 files reviewed at latest revision, 9 unresolved discussions.

---

*[vaadin-item.html, line 15 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIj5zJQKOSBI-O638t:-KxJdlxgflJHdNu3INJ_:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L15)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

@platosha  all those styles have been moved from vaadin-tab without modification, most of them were set there on this commit https://github.com/vaadin/vaadin-tabs/commit/9a2207fe  @jouni can you answer Anton ? 
</blockquote></details>

Done.

---

*[vaadin-item.html, line 16 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIjjwpHa8rC2ft3cFd:-KxJdmVgjseobAmGY8_F:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L16)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Same concern as above. Alignment settings for a flex container which does not make sense.
</blockquote></details>

Done.

---

*[vaadin-item.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIiaOQoj2vHt-7orMg:-KxJdn1q8cgjexcHyQT6:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L20)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Why `border-box` here?
</blockquote></details>

Done.

---

*[vaadin-item.html, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIbzkPJSktuzjXZR-e-r1-21:-KxJdnSTl3nTPXEvjJeT:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L21)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

- Remove display and align-items, let’s set those in the core/default styles (not the theme).
- Remove outline and opacity, which are only necessary when used inside a list element.
- Remove padding and box-sizing, let’s keep the default styles to bare minimum for the independent item.
- Remove flex-shrink, it’s only needed for `vaadin-tab`.
</blockquote></details>

Done.

---

*[vaadin-item.html, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIibmjyB5Z1PHNkans:-KxJdnorfWCeyv5C1Adi:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L21)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Duplicate of builtin style. FYI: the builtin style should probably be moved out too, see a comment below.
</blockquote></details>

Done.

---

*[vaadin-item.html, line 34 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIbzkPJSktuzjXZR-e-r1-34:-KxJdoO9psRNk4ZSWC3X:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L34)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Remove all state based styles.
</blockquote></details>

Done.

---

*[vaadin-item.html, line 43 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIkcNjAsWzMFk0Ng57:-KxJdos-Ouo2cponN_qx:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L43)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

OK probably does not affect much, but still have a question.

Default theme makes `display: flex`, which acts as a block for the parent.

Without default theme, it would be `inline-block`, which acts as an inline in the parent.

Why there is such a difference?
</blockquote></details>

Done.

---

*[vaadin-item.html, line 45 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIbzkPJSktuzjXZR-e-r1-45:-KxJdpO8YPPu5sEr2B7w:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L45)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Below: remove the disabled state related styles.
</blockquote></details>

Done.

---

*[vaadin-item.html, line 45 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIhdG2LhEPH_3-O9NW:-KxJdpoZ6vJhCMBGFPyJ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L45)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

OK maybe not a blocker, but not sure if this is appropriate here.

`flex-shrink` changes are only needed if a parent is a flex container.

If this is because of `<vaadin-tabs>` having a flex layout for slotted children, then should we rather move this fix to `<vaadin-tabs>` slotted styles?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5)*

**Comment by jouni** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5)*

**Comment by YuriyVaadin** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-:-KxMVnwEEEKtSSBo1Hgh:bnfp4nl)*

**Comment by platosha** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-item.html, line 15 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIj5zJQKOSBI-O638t:-KxMa7jGpeLMIh31gTON:bxat2jt) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L15)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

OK

I am still wondering about this `flex` and `align-items` here. Probably not necessary and might strike back.

Also, does not make sense much sense when reading in this context. None of the children are defined here, so why manipulate em...

But don’t want to block.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[vaadin-item.html, line 15 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-item/5#-KxIj5zJQKOSBI-O638t:-KxMbfC5LwuK8oNxtJbc:b-49x6fa) ([raw file](https://github.com/vaadin/vaadin-item/blob/a61b9c757832c56aeef2bb77fc787b485929112d/vaadin-item.html#L15)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

OK

I am still wondering about this `flex` and `align-items` here. Probably not necessary and might strike back.

Also, does not make sense much sense when reading in this context. None of the children are defined here, so why manipulate em...

But don’t want to block.
</blockquote></details>

I guess they are here because vaadin-item will have a bunch of reusable slots in the near future, @jouni should have a better explanation. 

Anyways merging this to unlock elements depending on this, styles could be improved in next versions

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5)*

**Comment by manolo** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-item/5)*

**Comment by jouni** [comment]
@platosha, about `flex` and `align-items`, a copy of my previous comment:

I’m not 100% sure if these are already good to have, but I’m expecting us to use flexbox once we add the planned slots to (icon/prefix, label/description and badge/action).

**Comment by platosha** [comment]
@jouni alright thanks for clarification!

**Pull request merged**